extensions/sql-migration/src/wizard/assessmentDetailsPage/instanceSummary.ts

```diff
@@ -0,0 +1,354 @@
+
+		// update width of states of ready, not ready and readyWithWarn in the graph bar component.
+		readinessStateIndex = 0;
+		let maxValue = 1;
+		readinessStates.forEach(readinessState => maxValue = Math.max(readinessState.count, maxValue));
+		this._divs.forEach(async (component) => {
+			const divWidth = this.setWidth(readinessStates[readinessStateIndex++].count ?? 0, maxValue);
+			await component.updateCssStyles({ 'width': divWidth + '%' });
+		});
+	}
+
+	// Create graph depicting dbs readiness.
+	private createGraphComponent(): azdata.FlexContainer {
+		const graph = this._view.modelBuilder.flexContainer().withProps({
+			CSSStyles: {
+				'display': 'flex',
+				'flex-direction': 'column',
+			}
+		}).component();
+
+		const readinessStates = [
+			{
+				label: constants.NOT_READY,
+				color: ColorCodes.NotReadyState_Red
+			},
+			{
+				label: constants.READY_WARN,
+				color: ColorCodes.ReadyWithWarningState_Amber
+			},
+			{
+				label: constants.READY,
+				color: ColorCodes.ReadyState_Green
+			}];
+
+		// create individual bar component for each property in above list
+		graph.addItems(readinessStates.map(l => this.createBarComponent(l)));
+		return graph;
+	}
+
+	// create individual bar component
+	private createBarComponent(linkMetaData: IActionMetadata) {
+
+		const barComponent = this._view.modelBuilder.flexContainer().withProps({
+			CSSStyles: {
+				'display': 'flex',
+				'flex-direction': 'row'
+			}
+		}).component();
+		barComponent.addItem(this.createYAxisComponent(), { 'flex': 'none' });
+
+		const division = this._view.modelBuilder.divContainer().withProps({
+			width: '0px',
+			CSSStyles: {
+				'margin-top': '12px',
+				'margin-left': '10px',
+				'background-color': linkMetaData.color ?? "",
+				'height': '30px',
+				'width': '0px'
+			}
+		}).component();
+
+		// storing divisions to update later based on selections.
+		this._divs.push(division);
+		barComponent.addItem(division);
+
+		return barComponent;
+	}
+
+	// create y axis details of the bar component
+	private createYAxisComponent(): azdata.FlexContainer {
+
+		const yAxisLabelContainer = this._view.modelBuilder.flexContainer().withProps({
+			CSSStyles: {
+				'width': '80px',
+				'margin-top': '12px',
+				'height': '30px',
+				'overflow-wrap': 'break-word'
+			}
+		}).component();
+
+		const yAxisLabel = this._view.modelBuilder.text().withProps({
+			value: "",
+			CSSStyles: {
+				...styles.CARD_AXES_LABEL,
+				'overflow-wrap': 'break-word'
+			},
+		}).component();
+
+		this._yAxisLabels.push(yAxisLabel);
+
+		yAxisLabelContainer.addItem(yAxisLabel);
+
+		return yAxisLabelContainer;
+	}
+
+	// maths formula to calculate width of bar component and create comparison among diff values.
+	private setWidth(value: number, maxValue: number): number {
+		const width = (Math.sqrt(value) / Math.sqrt(maxValue)) * 75;
```

Comment:
maxvalue should be checked 0

Reply:
it was ensured to be greater than 1 with min value assigned as `let maxValue = 1;`
but the function can be used be independently, thus adding a check in the function as well.